### ISSUE TITLE
Unable to modify font color in shapes

### ISSUE THREAD

**Issue by davidsclarke** [jmcnamara/excel-writer-xlsx#247]
According to the docs, the following code should make a workbook with a rectangle that has both a red outline, and red text.  It only produces a red outline.

Issue was reported to me (David Clarke) by phuah yee fei.  I have investigated this issue, and am initiating a bug request as the 1st step in submitting a patch.

```perl
my $workbook  = Excel::Writer::XLSX->new( 'red_rect.xlsx' );
my $worksheet = $workbook->add_worksheet();

my $format = $workbook->add_format(
    color => '#FF0000',
);

# Add a default rectangle shape.
my $rect = $workbook->add_shape(
    type => 'rect',
    text => "Hello\nWorld",
    format => $format,  # Text color red
    line => '#FF0000',   # Line color red
    fill => '#FFFFFF',   # Fill color white
);
$worksheet->insert_shape( 'B3', $rect );
```

**Comment by jmcnamara** [comment]
Thanks for the report. I'll have a loo.

**Comment by davidsclarke** [comment]
I have already looked into this, and it is a pretty minor fix.  I was planning on forking the dist, making the edits, and updated/new test scripts, and submitting a pull request back to you.  

One of my questions, is that the _get_palette_color sub that is in E-W-X-Shape, is now going to be identical to the sub of the same name in E-W-X-Worksheet.  Would you consider moving it to E-W-X-Utility, and exporting it.  

Would you rather me just send you a diff of what I changed?  I still use perl moderately, but little experience with GitHub, and no experience with Travis CI.

**Comment by jmcnamara** [comment]
> One of my questions, is that the _get_palette_color sub that is in E-W-X-Shape, is now going to be identical to the sub of the same name in E-W-X-Worksheet. Would you consider moving it to E-W-X-Utility, and exporting it

Maybe. Submit the patch first and if it make sense I'll refactor it.



> Would you rather me just send you a diff of what I changed? I still use perl moderately, but little experience with GitHub, and no experience with Travis CI.

Whatever works for you. :-) You can post the diff inline in your reply if you like.

**Comment by davidsclarke** [comment]
Here are the diff patches to fix (2 files) and test (1 file) this issue.    Let me know if you need anything else.  Note that the code below, duplicates the code for **_get_palette_color**, if you decide you want to refactor.

** lib/Shape.pm**

    147a147,149
    >         # Strip HTML style # prefix from colors if supplied
    >         $value =~ s/^#// if $key =~ m/(_fill|_line)/;
    > 
    175a178,180
    >         # Strip HTML style # prefix from colors if supplied
    >         $value =~ s/^#// if $key =~ m/(_fill|_line)/;
    > 
    264d268
    < # Note: This version doesn't add an alpha channel.
    271a276,280
    >     # Handle colours in #XXXXXX RGB format.
    >     if ( $index =~ m/^#([0-9A-F]{6})$/i ) {
    >         return "FF" . uc( $1 );
    >     }
    > 
    278c287
    <     return sprintf "%02X%02X%02X", @rgb[0, 1, 2];
    ---
    >     return sprintf "FF%02X%02X%02X", @rgb[0, 1, 2];
    281a291
    > 
    371c381
    <     $shape->set_properties( type => 'ellipse', text => 'Hello' );
    ---
    >     $shape->set_properties( type => 'ellipse', text => 'Hello', format => $format );
    493c503
    < Workbook format for decorating the shape text (font family, size, and decoration).
    ---
    > Workbook format for decorating the shape text (font family, size, color, and decoration).
    522c532
    < Shape colour for the outline and fill. Colours may be specified as a colour index, or in RGB format, i.e. C<AA00FF>.
    ---
    > Shape colour for the outline and fill. Colours may be specified as a colour index, or in HTML format, i.e. C<#AA00FF>.
    595a606,610
    >     # Add a format for the text color
    >     my $format = $workbook->add_format(
    >       color => '#FF0000',
    > );

**lib/Drawing.pm**

    1442c1442
    <         $color =~ s/^FF//;    # Remove leading FF from rgb for shape color.
    ---
    >         $color =~ s/^FF// if $color =~ m/^FF[0-9A-F]{6}$/i;    # Remove leading FF from rgb for shape color.

**t/drawing/drawing_shape_04.t**
    9c9
    < use TestFunctions qw(_expected_to_aref _got_to_aref _is_deep_diff _new_object);
    ---
    > use TestFunctions qw(_expected_to_aref _got_to_aref _is_deep_diff _new_object _new_workbook);
    27c27,29
    < my $shape = Excel::Writer::XLSX::Shape->new( undef, text => 'test', id => 1000 );
    ---
    > my $wbk = _new_workbook( \$got2 );
    > my $format = $wbk->add_format(color => '#FF0000', size => 8);
    > my $shape = Excel::Writer::XLSX::Shape->new( undef, text => 'test', format=>$format, id => 1000 );
    114c116
    <                                 <a:srgbClr val="000000"/>
    ---
    >                                 <a:srgbClr val="FF0000"/>

**Comment by jmcnamara** [comment]
Just getting around to having a look at this. Could you add the output from git diff (or show or whatever) for this. The above is a little hard to parse.

**Comment by jmcnamara** [comment]
Closing until there is a patch or Pull Request.
